src/sys/mod.rs

```diff
@@ -13,52 +13,81 @@
+            $($(#[$target: meta])* $libc: ident :: $flag: ident),+ $(,)*
+        ) => {
+            struct $type($event_type);
 
-        impl fmt::Debug for $type {
-            fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
-                let mut written_one = false;
-                $(
-                    $(#[$target])*
-                    #[allow(clippy::bad_bit_mask)] // Apparently some flags are zero.
-                    {
-                        // Windows doesn't use `libc` but the `afd` module.
-                        if $test(&self.0, &$libc :: $flag) {
-                            if !written_one {
-                                write!(f, "{}", stringify!($flag))?;
-                                written_one = true;
-                            } else {
-                                write!(f, "|{}", stringify!($flag))?;
+            impl fmt::Debug for $type {
+                fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+                    let mut written_one = false;
+                    $(
+                        $(#[$target])*
+                        #[allow(clippy::bad_bit_mask)] // Apparently some flags are zero.
+                        {
+                            // Windows doesn't use `libc` but the `afd` module.
+                            if $test(&self.0, &$libc :: $flag) {
+                                if !written_one {
+                                    write!(f, "{}", stringify!($flag))?;
+                                    written_one = true;
+                                } else {
+                                    write!(f, "|{}", stringify!($flag))?;
+                                }
                             }
                         }
+                    )+
+                    if !written_one {
+                        write!(f, "(empty)")
+                    } else {
+                        Ok(())
                     }
-                )+
-                if !written_one {
-                    write!(f, "(empty)")
-                } else {
-                    Ok(())
                 }
             }
-        }
-    };
+        };
+    }
 }
 
 #[cfg(unix)]
-pub use self::unix::{
-    event, Event, Events, Selector, SocketAddr, SourceFd, TcpListener, TcpStream, UdpSocket,
-    UnixDatagram, UnixListener, UnixStream, Waker,
-};
+cfg_os_poll! {
+    mod unix;
+    pub use self::unix::SourceFd;
+    pub(crate) use self::unix::{event, Event, Events, Selector, Waker};
 
-#[cfg(unix)]
-mod unix;
+    cfg_tcp! {
+        pub(crate) use self::unix::{TcpListener, TcpStream};
+    }
 
-#[cfg(windows)]
-pub use self::windows::{event, Event, Events, Selector, TcpListener, TcpStream, UdpSocket, Waker};
+    cfg_udp! {
+        pub(crate) use self::unix::UdpSocket;
+    }
+
+    cfg_uds! {
+        pub(crate) use self::unix::{UnixDatagram, UnixListener, UnixStream};
+        pub use self::unix::SocketAddr;
```

Comment:
Is there are reason for the mix of `pub` and `pub(crate)`, I think can all be `pub(crate)`?

Reply:
`SocketAddr` should be `pub` because it's the type returned by `UnixListener::accept`. The other resources should only be used within the crate because they are private fields in the `net` equivalents.